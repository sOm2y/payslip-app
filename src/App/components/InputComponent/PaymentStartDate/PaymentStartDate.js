import React from 'react';
import TextField from 'material-ui/TextField';
import { FormControl } from 'material-ui/Form';

import generateDateRange from '../../../../shared/generateDateRange';
import { renderFailText } from '../shared';

const dates = generateDateRange();

class PaymentStartDate extends React.PureComponent<InputProps> {
  render() {
    const { onChange, value, failReasons } = this.props;

    return (
      <FormControl error={!!failReasons && failReasons.length !== 0} style={{ width: '100%' }}>
        <TextField
          SelectProps={{
            native: true
          }}
          select
          label="Payment Start Date"
          value={value}
          onChange={(evt) => onChange(evt.target.value)}
          margin="normal"
          error={!!failReasons && failReasons.length !== 0}
        >
          <option value={-1}> </option>
          {dates.map((date) => (
            <option key={date.month} value={date.month}>
              {date.range}
            </option>
          ))}
        </TextField>
        {failReasons && renderFailText(failReasons)}
      </FormControl>
    );
  }
}

export default PaymentStartDate;
