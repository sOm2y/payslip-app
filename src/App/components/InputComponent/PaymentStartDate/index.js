//@flow

export { default } from './PaymentStartDate';
