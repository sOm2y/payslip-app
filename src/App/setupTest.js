//@flow

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// enable using of expect(wrapper).toContainReact(<SomeComponent/>)
import 'jest-enzyme';

configure({ adapter: new Adapter() });
