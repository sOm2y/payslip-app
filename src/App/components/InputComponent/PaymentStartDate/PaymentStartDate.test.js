//@flow

import React from 'react';
import PaymentStartDate from './PaymentStartDate';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import '../../../setupTest'; // enzyme setup

describe('PaymentStartDate', () => {
  it('should render correctly, and maches snapshot', () => {
    const tree = renderer.create(<PaymentStartDate />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render fail reasons correctly', () => {
    const reasons = ['oops'];

    const sut = mount(<PaymentStartDate failReasons={[]} />);

    sut.setProps({ failReasons: reasons });

    expect(sut.html()).toContain(reasons[0]);
  });

  it('should response to select changes', () => {
    const onChangeStub = jest.fn();

    const sut = mount(<PaymentStartDate onChange={onChangeStub} />);

    sut.find('select').simulate('change', { target: { value: '0', focus: () => {} } });

    expect(onChangeStub).toBeCalledWith('0');
  });
});
