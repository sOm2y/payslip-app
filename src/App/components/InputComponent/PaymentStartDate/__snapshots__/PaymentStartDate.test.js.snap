// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentStartDate should render correctly, and maches snapshot 1`] = `
<div
  className="MuiFormControl-root-1"
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
    onBlur={[Function]}
    onFocus={[Function]}
    required={false}
  >
    <label
      className="MuiFormLabel-root-11 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
      data-shrink={false}
      htmlFor={undefined}
    >
      Payment Start Date
    </label>
    <div
      className="MuiInput-root-20 MuiInput-formControl-21 MuiInput-inkbar-22 MuiInput-underline-28"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiSelect-root-15"
      >
        <select
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiSelect-select-16 MuiInput-input-24 MuiInput-inputSingleline-31"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name={undefined}
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value={undefined}
        >
          <option
            value={-1}
          >
             
          </option>
          <option
            value="10"
          >
            1st November - 30th November, 2017
          </option>
          <option
            value="11"
          >
            1st December - 31st December, 2017
          </option>
          <option
            value="0"
          >
            1st January - 31st January, 2018
          </option>
          <option
            value="1"
          >
            1st February - 28th February, 2018
          </option>
          <option
            value="2"
          >
            1st March - 31st March, 2018
          </option>
          <option
            value="3"
          >
            1st April - 30th April, 2018
          </option>
          <option
            value="4"
          >
            1st May - 31st May, 2018
          </option>
          <option
            value="5"
          >
            1st June - 30th June, 2018
          </option>
          <option
            value="6"
          >
            1st July - 31st July, 2018
          </option>
          <option
            value="7"
          >
            1st August - 31st August, 2018
          </option>
          <option
            value="8"
          >
            1st September - 30th September, 2018
          </option>
          <option
            value="9"
          >
            1st October - 31st October, 2018
          </option>
        </select>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-35 MuiSelect-icon-19"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
